client/pages/live-map.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { Map, GoogleApiWrapper, Marker } from 'google-maps-react';
+
+class LiveMap extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dailyRoutes: [],
+      maxIncrease: 0,
+      totalVehicles: 0,
+      count: 0,
+      mapCenter: {
+        lat: 25.774,
+        lng: -80.190
+      },
+      vehicles: [{
+        vehicleId: 1,
+        coords: {
+          lat: 25.774,
+          lng: -80.190
+        }
+      }, {
+        vehicleId: 2,
+        coords: {
+          lat: 24.774,
+          lng: -80.190
+        }
+      }]
+    };
+    fetch('/api/dailyroutes')
+      .then(res => res.json())
+      .then(result => {
+
+        let routesArr = [];
+        const dailyRoutesArr = [];
+        this.setState({ totalVehicles: result.rows.length });
+        this.setState({ maxIncrease: result.rows[0].dailyRoute.gpx.trk[0].trkseg[0].trkpt.length });
+        for (let i = 0; i < result.rows.length; i++) {
+          for (let j = 0; j < result.rows[0].dailyRoute.gpx.trk[0].trkseg[0].trkpt.length; j++) {
+            routesArr.push(result.rows[i].dailyRoute.gpx.trk[0].trkseg[0].trkpt[j].ATTR);
+
+          }
+
+          dailyRoutesArr.push(routesArr);
+          routesArr = [];
+        }
+        this.setState({ dailyRoutes: dailyRoutesArr });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+
+  }
+
+  count() {
+    let increase = this.state.count;
+
+    setInterval(() => {
+
+      increase++;
+      this.setState({ count: increase });
+      const maxIncrease = this.state.maxIncrease;
+
+      const totalVehicles = this.state.vehicles.length;
+
+      for (let i = 0; i < totalVehicles; i++) {
+        const newCoords = this.state.vehicles[i];
+
+        newCoords.coords.lat = this.state.dailyRoutes[i][increase].lat;
+        newCoords.coords.lng = this.state.dailyRoutes[i][increase].lon;
+
+        this.setState(newCoords);
+      }
```

Comment:
This accidentally works. Instead of mutating your original vehicles, try calculating new vehicles with `.map()`

```js
const updatedVehicles = this.state.vehicles.map((vehicle, index) => {
  return Object.assign({}, vehicle, { coords: this.state.dailyRoutes[index][increase] })
})
this.setState({ vehicles: updatedVehicles })
```